packages/react-accordion/src/components/AccordionHeader/useAccordionHeader.tsx

```diff
@@ -86,3 +92,16 @@ export const useAccordionHeader = (props: AccordionHeaderProps, ref: React.Ref<H
     }),
   };
 };
+
+function mapStateToRotation({
+  open,
+  expandIconPosition,
+}: Pick<AccordionHeaderContextValue, 'open' | 'expandIconPosition'>) {
+  if (open && expandIconPosition === 'end') {
+    return '-90';
+  }
+  if ((!open && expandIconPosition === 'end') || (open && expandIconPosition === 'start')) {
+    return '90';
+  }
+  return '0';
+}
```

Comment:
This function was copied verbatim from (now-deleted) AccordionHeaderExpandIcon.tsx